### PR TITLE
[17.0] [MIG] product_multi_company: Migration to 17.0

### PR DESCRIPTION
**Pull request by rven**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@rven can you rebase and fix conflicts?

**Review by chienandalu — COMMENTED** [review]
Can you rebase your migration over branch 16.0?

**Comment by rven** [comment]
> Can you rebase your migration over branch 16.0?

Why 16? this is the 17.0 migration...

**Comment by pedrobaeza** [comment]
"Rebase" means to check the latest commits from 16.0 to be incorporated here, which there are several.

File: product_multi_company/tests/test_product_multi_company.py [hunk 1]
```diff
@@ -11,7 +11,6 @@
 class TestProductMultiCompany(ProductMultiCompanyCommon, common.TransactionCase):
     def test_create_product(self):
         product = self.env["product.product"].create({"name": "Test"})
-        self.assertFalse(product.company_id)
```
Comment (pedrobaeza) [inline product_multi_company/tests/test_product_multi_company.py]: Why this line is being removed? This means a change of behavior.

Comment (rven) [reply product_multi_company/tests/test_product_multi_company.py]: Previously it was giving errors, now it isn't anymore

**Review by pedrobaeza — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]
Code and functional review :+1:

**Pull request merged**